olive/systems/local.py

```diff
@@ -31,20 +31,12 @@ def run_pass(
         """
         return the_pass.run(model, output_model_path, point)
 
-    def evaluate_model(self, model: OliveModel, metrics: List[Metric]) -> Dict[str, Any]:
+    def evaluate_model(self, model: OliveModel, metrics: List[Metric], device: Device = Device.CPU) -> Dict[str, Any]:
         """
         Evaluate the model
         """
-
         if isinstance(model, CompositeOnnxModel):
             raise NotImplementedError()
 
-        metrics_res = {}
-        for metric in metrics:
-            if metric.type == MetricType.ACCURACY:
-                metrics_res[metric.name] = evaluate_accuracy(model, metric, self.device)
-            elif metric.type == MetricType.LATENCY:
-                metrics_res[metric.name] = evaluate_latency(model, metric, self.device)
-            else:
-                metrics_res[metric.name] = evaluate_custom_metric(model, metric, self.device)
-        return metrics_res
+        evaluator: OliveEvaluator = OliveEvaluatorFactory.create_evaluator_for_model(model)
+        return evaluator.evaluate(model, metrics, device=device)
```

Comment:
Like in my previous comment, can you pass `device=self.device`?

Reply:
Done!